pkg/scalers/azure_eventhub.go

```diff
@@ -0,0 +1,127 @@
+package scalers
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+
+	eventhub "github.com/Azure/azure-event-hubs-go"
+	"github.com/Azure/azure-storage-blob-go/azblob"
+	"github.com/Azure/go-autorest/autorest/azure"
+)
+
+const (
+	environmentName = "AzurePublicCloud"
+)
+
+type Checkpoint struct {
+	Epoch          int64  `json:"Epoch"`
+	Offset         string `json:"Offset"`
+	Owner          string `json:"Owner"`
+	PartitionID    string `json:"PartitionId"`
+	SequenceNumber int64  `json:"SequenceNumber"`
+	Token          string `json:"Token"`
+}
+
+// GetStorageCredentials returns azure env and storage credentials
+func GetStorageCredentials(storageConnection string) (azure.Environment, *azblob.SharedKeyCredential, error) {
+	storageAccountName, storageAccountKey, err := ParseAzureStorageConnectionString(storageConnection)
+	if err != nil {
+		return azure.Environment{}, &azblob.SharedKeyCredential{}, fmt.Errorf("unable to parse connection string: %s", storageConnection)
+	}
+
+	azureEnv, err := azure.EnvironmentFromName(environmentName)
+	if err != nil {
+		return azureEnv, nil, fmt.Errorf("could not get azure.Environment struct: %s", err)
+	}
+
+	cred, err := azblob.NewSharedKeyCredential(storageAccountName, storageAccountKey)
+	if err != nil {
+		return azureEnv, nil, fmt.Errorf("could not prepare a blob storage credential: %s", err)
+	}
+
+	return azureEnv, cred, nil
+}
+
+// GetEventHubClient returns eventhub client
+func GetEventHubClient(connectionString string) (*eventhub.Hub, error) {
+	hub, err := eventhub.NewHubFromConnectionString(connectionString)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create hub client: %s", err)
+	}
+
+	return hub, nil
+}
+
+// GetCheckpointFromBlobStorage accesses Blob storage and gets checkpoint information of a partition
+func GetCheckpointFromBlobStorage(ctx context.Context, partitionID string, eventHubMetadata EventHubMetadata) (Checkpoint, error) {
+	storageAccountName, _, err := ParseAzureStorageConnectionString(eventHubMetadata.storageConnection)
+	if err != nil {
+		return Checkpoint{}, fmt.Errorf("unable to parse storage connection string: %s", err)
+	}
+
+	eventHubNamespace, eventHubName, err := ParseAzureEventHubConnectionString(eventHubMetadata.eventHubConnection)
+	if err != nil {
+		return Checkpoint{}, fmt.Errorf("unable to parse event hub connection string: %s", err)
+	}
+
+	// TODO: add more ways to read from different types of storage and read checkpoints/leases written in different JSON formats
+	u, _ := url.Parse(fmt.Sprintf("https://%s.blob.core.windows.net/azure-webjobs-eventhub/%s/%s/%s/%s", storageAccountName, eventHubNamespace, eventHubName, eventHubMetadata.eventHubConsumerGroup, partitionID))
```

Comment:
Super low-priority, but you might also consider supporting the local Azure Storage emulator connection string format, which has far less information in it but is useful for local development and testing.